model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/embedded/CacheConfigurator.java

```diff
@@ -507,6 +508,12 @@ public static ConfigurationBuilder getCacheConfiguration(String cacheName, boole
             case OFFLINE_USER_SESSION_CACHE_NAME:
                 if (clustered) {
                     builder.clustering().cacheMode(CacheMode.DIST_SYNC).hash().numOwners(1);
+
```

Comment:
I would disable the state transfer in the `configureSessionsCachesForVolatileSessions` and `configureSessionsCachesForPersistentSessions` instead. 

After these lines:
https://github.com/keycloak/keycloak/blob/ca617d971156ea3d457b6bbfacf3191030a003c5/model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/embedded/CacheConfigurator.java#L255
https://github.com/keycloak/keycloak/blob/ca617d971156ea3d457b6bbfacf3191030a003c5/model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/embedded/CacheConfigurator.java#L306

Reply:
Thanks, that's much nicer and I hadn't concerned the offline caches when !persistent-user-sessions.